massa-api/src/public.rs

```diff
@@ -580,40 +580,47 @@ impl MassaRpcServer for API<Public> {
         Ok(res)
     }
 
-    /// gets a block. Returns None if not found
+    /// gets a block(s). Returns nothing if not found
     /// only active blocks are returned
-    async fn get_block(&self, id: BlockId) -> RpcResult<BlockInfo> {
+    async fn get_blocks(&self, ids: Vec<BlockId>) -> RpcResult<Vec<BlockInfo>> {
         let consensus_controller = self.0.consensus_controller.clone();
         let storage = self.0.storage.clone_without_refs();
-        let block = match storage.read_blocks().get(&id).cloned() {
-            Some(b) => b.content,
-            None => {
-                return Ok(BlockInfo { id, content: None });
-            }
-        };
-
-        let graph_status = consensus_controller
-            .get_block_statuses(&[id])
+        let blocks = ids
             .into_iter()
-            .next()
-            .expect("expected get_block_statuses to return one element");
-
-        let is_final = graph_status == BlockGraphStatus::Final;
-        let is_in_blockclique = graph_status == BlockGraphStatus::ActiveInBlockclique;
-        let is_candidate = graph_status == BlockGraphStatus::ActiveInBlockclique
-            || graph_status == BlockGraphStatus::ActiveInAlternativeCliques;
-        let is_discarded = graph_status == BlockGraphStatus::Discarded;
-
-        Ok(BlockInfo {
-            id,
-            content: Some(BlockInfoContent {
-                is_final,
-                is_in_blockclique,
-                is_candidate,
-                is_discarded,
-                block,
-            }),
-        })
+            .filter_map(|id| {
+                let block = if let Some(wrapped_block) = storage.read_blocks().get(&id) {
```

Comment:
I would suggest to rewrite it like that (so it will be easier to read):

```

let block = if let Some(...) = ... {
     ....
     if let Some(graph_status) = consensus_controller... {
         ...
        return Some(BlockInfoContent { ... });
    }
}
return None;



```

Reply:
Done